### PR TITLE
[MM-39918]: Add ability to list archived private channels with sys admin permission

### PR DESCRIPTION
**Pull request by haardikdharma10**
#### Summary

This PR adds ability for a system admin to list archived private channels regardless of their membership

#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/19913

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by noxer — APPROVED** [review]
LGTM

File: app/channel.go [hunk 1]
```diff
@@ -1912,8 +1912,15 @@ func (a *App) GetAllChannelsCount(c request.CTX, opts model.ChannelSearchOpts) (
 	return count, nil
 }
 
-func (a *App) GetDeletedChannels(c request.CTX, teamID string, offset int, limit int, userID string) (model.ChannelList, *model.AppError) {
-	list, err := a.Srv().Store.Channel().GetDeleted(teamID, offset, limit, userID)
+func (a *App) GetDeletedChannels(c request.CTX, teamID string, offset int, limit int, userID string, isAdmin bool) (model.ChannelList, *model.AppError) {
```
Comment (noxer) [inline app/channel.go:1915]: You'll need to adapt the tests for this to pass.

**Review by noxer — CHANGES_REQUESTED** [review]

File: app/channel.go (hunk 1 above)
Comment (haardikdharma10) [reply app/channel.go:1915]: Hey @noxer, I have updated the tests accordingly.

Comment (noxer) [reply app/channel.go:1915]: Perfect, thanks.

Comment (isacikgoz) [inline app/channel.go:1915]: Let's document the method what `isAdmin` makes difference.

**Review by isacikgoz — COMMENTED** [review]
Looks good, @haardikdharma10 should we also add specific tests to this?

**Comment by isacikgoz** [comment]
Hey @haardikdharma10 let us know if you need any guidance with our comments :)

**Comment by noxer** [comment]
Closing due to inactivity.

**Pull request closed without merging**